### PR TITLE
Unify supported runtime fields script contexts

### PR DESCRIPTION
**Pull request by javanna**
There's a few places where we need to access all of the supported runtime fields script contexts. Up until now we have listed them in all those places, but a better way would be to have them listed in one place and access that same list from all consumers. This is what this commit introduces.

Along with the introduction of runtime fields contexts in ScriptModule, we rename the whitelist files so that they contain their corresponding context name to simplify looking them up.

### CODE REVIEW

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java [hunk 1]
```diff
@@ -110,18 +113,18 @@ private PainlessExecuteAction() {
...
+            contexts.put(PainlessTestScript.CONTEXT.name, PainlessTestScript.CONTEXT);
+            contexts.put(FilterScript.CONTEXT.name, FilterScript.CONTEXT);
+            contexts.put(ScoreScript.CONTEXT.name, ScoreScript.CONTEXT);
+            for (ScriptContext<?> runtimeFieldsContext : ScriptModule.RUNTIME_FIELDS_CONTEXTS) {
+                contexts.put(runtimeFieldsContext.name, runtimeFieldsContext);
+            }
+            return Collections.unmodifiableMap(contexts);
+        }
+
+        static final Map<String, ScriptContext<?>> SUPPORTED_CONTEXTS = getSupportedContexts();
```
Comment (nik9000) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/action/PainlessExecuteAction.java:127]: I tend to init stuff like this in a static block. But this is cool too.

File: modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessPlugin.java [hunk 2]
```diff
@@ -96,13 +90,10 @@
...
-        map.put(BooleanFieldScript.CONTEXT, getRuntimeFieldWhitelist("boolean"));
-        map.put(DateFieldScript.CONTEXT, getRuntimeFieldWhitelist("date"));
-        map.put(DoubleFieldScript.CONTEXT, getRuntimeFieldWhitelist("double"));
-        map.put(LongFieldScript.CONTEXT, getRuntimeFieldWhitelist("long"));
-        map.put(StringFieldScript.CONTEXT, getRuntimeFieldWhitelist("string"));
-        map.put(GeoPointFieldScript.CONTEXT, getRuntimeFieldWhitelist("geopoint"));
-        map.put(IpFieldScript.CONTEXT, getRuntimeFieldWhitelist("ip"));
+
+        for (ScriptContext<?> scriptContext : ScriptModule.RUNTIME_FIELDS_CONTEXTS) {
+            map.put(scriptContext, getRuntimeFieldWhitelist(scriptContext.name));
```
Comment (nik9000) [inline modules/lang-painless/src/main/java/org/elasticsearch/painless/PainlessPlugin.java:95]: :+1:

**Review by nik9000 — APPROVED** [review]
Looks like good tidying up to me.

**Review by jdconrad — APPROVED** [review]
LGTM. Thanks for cleaning this up.

**Pull request merged**